test/requestContextPlugin.e2e.spec.js

```diff
@@ -0,0 +1,106 @@
+const fastify = require('fastify')
+const request = require('superagent')
+const { fastifyRequestContextPlugin } = require('../lib/requestContextPlugin')
+const { TestService } = require('./internal/testService')
+
+function initAppPostWithPrevalidation(endpoint) {
+  return new Promise((resolve) => {
```

Comment:
Do not wrap with a promise, use an async function. `.listen()` returns a promise if a callback is not passed in.

Reply:
Done